servicetest/environment/test_environment.py

```diff
@@ -0,0 +1,276 @@
+
+    def test_setting_same_var_keeps_initial_value(self):
+        """ Setting a variable twice should result first value being set and kept.
+        """
+        try:
+            sc = Container()
+            sc.start(DATA)
+            # Set once
+            sc.set_gateway_config("env", GW_CONFIG_ONE_VAR)
+            # Set twice
+            sc.set_gateway_config("env", GW_CONFIG_SAME_VAR)
+            sc.launch_command("python " +
+                              sc.get_bind_dir() +
+                              "/testhelper.py --get-env-vars " +
+                              sc.get_bind_dir() + "/testoutput/")
+
+            # The D-Bus LaunchCommand is asynch so let it take effect before assert
+            time.sleep(0.5)
+            my_env_var = testhelper.env_var("MY_ENV_VAR", TESTOUTPUT_DIR)
+            # Assert value is the one set first
+            assert my_env_var == "1234"
+        finally:
+            sc.terminate()
+
+    def test_appending_then_adding_keeps_initial_value(self):
+        """ Append a non existing variable. Then set it per the normal way.
+            Assert the value created when "appending" non existing value
+            is kept intact.
+        """
+        try:
+            sc = Container()
+            sc.start(DATA)
+            # Append a variable not set before
+            sc.set_gateway_config("env", GW_CONFIG_APPEND)
+            # Set the variable that was "appended" previously
+            sc.set_gateway_config("env", GW_CONFIG_TWO_VARS)
+            sc.launch_command("python " +
+                              sc.get_bind_dir() +
+                              "/testhelper.py --get-env-vars " +
+                              sc.get_bind_dir() + "/testoutput/")
+
+            # The D-Bus LaunchCommand is asynch so let it take effect before assert
+            time.sleep(0.5)
+            ld_library_path = testhelper.env_var("LD_LIBRARY_PATH", TESTOUTPUT_DIR)
+            # Assert the "appended" value is kept
+            assert ld_library_path == ":/another/path"
+        finally:
+            sc.terminate()
+
+    def test_appending_non_existing_var_creates_it(self):
+        """ Append to a non existing env var and assert that it gets created
+            instead.
+        """
+        try:
+            sc = Container()
+            sc.start(DATA)
+            # Append a variable not set before
+            sc.set_gateway_config("env", GW_CONFIG_APPEND)
+            sc.launch_command("python " +
+                              sc.get_bind_dir() +
+                              "/testhelper.py --get-env-vars " +
+                              sc.get_bind_dir() + "/testoutput/")
+
+            # The D-Bus LaunchCommand is asynch so let it take effect before assert
+            time.sleep(0.5)
+            variable = testhelper.env_var("LD_LIBRARY_PATH", TESTOUTPUT_DIR)
+            # Assert the append resultet in the creation of the variable
+            assert variable == ":/another/path"
```

Comment:
I will try to clarify the method comment a bit. I'd rather not explain it again in the comment above the actual assert, as the tests comment itself says that this will happen, the comment above the assert I thought should just clarify what is asserted, not the whole test.